Sources/SwiftSyntaxMacroExpansion/MacroSystem.swift

```diff
@@ -382,6 +385,30 @@ class MacroApplication<Context: MacroExpansionContext>: SyntaxRewriter {
   }
 
   // Subscripts
+
+  // Declarations with optional trailing code blocks.
+  func visit<DeclType: DeclSyntaxProtocol & HasTrailingOptionalCodeBlock>(
+    function: DeclType
+  ) -> DeclSyntax {
+    let body = expandBody(decl: function).map { visit($0) }
+    return DeclSyntax(function.with(\.body, body))
+  }
+
+  override func visit(_ node: AccessorDeclSyntax) -> DeclSyntax {
+    return visit(function: node)
+  }
+
+  override func visit(_ node: DeinitializerDeclSyntax) -> DeclSyntax {
+    return visit(function: node)
+  }
+
+  override func visit(_ node: FunctionDeclSyntax) -> DeclSyntax {
+    return visit(function: node)
+  }
+
+  override func visit(_ node: InitializerDeclSyntax) -> DeclSyntax {
+    return visit(function: node)
+  }
```

Comment:
It seems a little fragile that we’ll forget to add a new `visit` function here if there’s a new node that conforms to `HasTrailingOptionalCodeBlock`. Wouldn’t it be better to do the following in `visitAny` or am I missing something?

```swift
if let  node = node.as(SyntaxProtocol.self) as DeclSyntaxProtocol & HasTrailingOptionalCodeBlock {
  visit(node)
}
```

Reply:
Yes, I like that